apps/prairielearn/src/ee/pages/institutionAdminCourse/institutionAdminCourse.ts

```diff
@@ -39,4 +43,52 @@ router.get(
   }),
 );
 
+const UpdateEnrollmentLimitsBodySchema = z.object({
+  __action: z.literal('update_enrollment_limits'),
+  yearly_enrollment_limit: z.union([z.literal(''), z.coerce.number().int()]),
+  course_instance_enrollment_limit: z.union([z.literal(''), z.coerce.number().int()]),
+});
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const course = await queryRow(
+      sql.select_course,
+      {
+        institution_id: req.params.institution_id,
+        course_id: req.params.course_id,
+      },
+      CourseSchema,
+    );
+
+    if (req.body.__action === 'update_enrollment_limits') {
+      const body = UpdateEnrollmentLimitsBodySchema.parse(req.body);
```

Comment:
Almost certainly a premature optimization, but it felt better to avoid re-allocating a schema on every request. Maybe this just doesn't matter though?

Reply:
I can't imagine that this would even be a measurable cost, compared to everything else we are doing to handle the request. I would prioritize code readability over this kind of micro-(nano?)-optimization.

Reply:
Entirely fair! Done in f2e766f.